### PR TITLE
first crack at a worker pool for s3 and sqs/sns

### PR DESCRIPTION
**Pull request by david-fritz-gravwell**

### CODE REVIEW

File: ingesters/s3Ingester/config.go [hunk 1]
```diff
@@ -95,6 +97,7 @@ func GetConfig(path, overlayPath string) (*cfgType, error) {
 		IngestConfig:         cr.Global.IngestConfig,
 		Attach:               cr.Attach,
 		State_Store_Location: cr.Global.State_Store_Location,
+		Worker_Pool_Size:     cr.Global.Worker_Pool_Size,
```
Comment (kris-watts-gravwell) [inline ingesters/s3Ingester/config.go:100]: init cr.Global.Worker_Pool_Size to one prior to reading the file into the cr structure.

**Review by kris-watts-gravwell — CHANGES_REQUESTED** [review]
init pool to 1 prior to reading in the config file.

File: ingesters/s3Ingester/manager.go [hunk 2]
```diff
@@ -19,28 +19,37 @@ import (
...
 	return
 }
 
-func sqsS3Routine(s *SQSS3Listener, wg *sync.WaitGroup, ctx context.Context, lg *log.Logger) {
+func sqsS3Routine(s *SQSS3Listener, wg *sync.WaitGroup, ctx context.Context, lg *log.Logger, numWorkers int) {
 	defer wg.Done()
 
+	// create workers
+	var workerWg sync.WaitGroup
+	queue := make(chan *sqs.Message, QUEUE_DEPTH)
```
Comment (kris-watts-gravwell) [inline ingesters/s3Ingester/manager.go:47]: I don't see that this ever actually gets closed?  how are workers shutdown on this code path?

**Review by kris-watts-gravwell — CHANGES_REQUESTED** [review]
On the normal code path i don't see anything that looks like it closes the input queue

File: ingesters/s3Ingester/manager.go (hunk 2 above)
Comment (david-fritz-gravwell) [reply ingesters/s3Ingester/manager.go:47]: good catch -- PTAL

**Review by kris-watts-gravwell — APPROVED** [review]
Lgtm.  But let's test this on gin before we release to the world

**Pull request merged**